src/lib.rs

```diff
@@ -885,51 +925,747 @@ impl GooseAttack {
+    fn set_throttle_requests(&mut self) -> Result<(), GooseError> {
+        self.throttle_requests = if let Some(throttle_requests) =
+            self.configuration.throttle_requests
+        {
+            // Don't allow --throttle-requests with --manager.
+            if self.attack_mode == GooseMode::Manager {
+                return Err(GooseError::InvalidOption {
+                    option: "--throttle-requests".to_string(),
+                    value: self.configuration.throttle_requests.unwrap().to_string(),
+                    detail: "The --throttle-requests option can not be set together with the --manager flag.".to_string(),
+                });
+            }
+            // Be sure throttle_requests is in allowed range.
+            if throttle_requests == 0 {
+                return Err(GooseError::InvalidOption {
+                    option: "--throttle-requests".to_string(),
+                    value: throttle_requests.to_string(),
+                    detail: "The --throttle-requests option must be set to at least 1 request per second.".to_string(),
+                });
+            } else if throttle_requests > 1_000_000 {
+                return Err(GooseError::InvalidOption {
+                    option: "--throttle-requests".to_string(),
+                    value: throttle_requests.to_string(),
+                    detail: "The --throttle-requests option can not be set to more than 1,000,000 requests per second.".to_string(),
+                });
+            }
+            throttle_requests
+        } else if let Some(throttle_requests) = self.defaults.throttle_requests {
+            if self.attack_mode == GooseMode::Manager {
+                0
+            } else {
+                // Be sure throttle_requests is in allowed range.
+                if throttle_requests == 0 {
+                    return Err(GooseError::InvalidOption {
+                        option: "GooseDefault::ThrottleRequests".to_string(),
+                        value: throttle_requests.to_string(),
+                        detail: "The GooseDefault::ThrottleRequests default must be set to at least 1 request per second.".to_string(),
+                    });
+                } else if throttle_requests > 1_000_000 {
+                    return Err(GooseError::InvalidOption {
+                        option: "GooseDefault::ThrottleRequests".to_string(),
+                        value: throttle_requests.to_string(),
+                        detail: "The GooseDefault::ThrottleRequests default can not be set to more than 1,000,000 requests per second.".to_string(),
+                    });
+                }
+                throttle_requests
+            }
+        } else {
+            0
+        };
+
+        if self.throttle_requests > 0 {
+            info!("throttle_requests = {}", self.throttle_requests);
+        }
+
+        Ok(())
+    }
+
+    // Determine if no_reset_statics is enabled.
+    fn no_reset_metrics(&self) -> bool {
```

Comment:
@LionsAd should this be named `get_no_reset_metrics`? We're just determining if this flag is set (or defaulted to `true`)

Reply:
updated to `set_no_reset_metrics()` which updates `configuration` with default if set